kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalSchemaRegistry.java

```diff
@@ -1,18 +1,34 @@
 package com.provectus.kafka.ui.model;
 
-import java.util.List;
 import lombok.Builder;
 import lombok.Data;
 
+import java.util.List;
+
 @Data
-@Builder(toBuilder = true)
 public class InternalSchemaRegistry {
   private final String username;
   private final String password;
   private final List<String> url;
+  private final FailoverUrlList urlList;
+
+  @Builder(toBuilder = true)
+  public InternalSchemaRegistry(String username, String password, List<String> url) {
+    this.username = username;
+    this.password = password;
+    this.url = url;
+    this.urlList = new FailoverUrlList(url);
+  }
+
+  public String getUri() {
+    return urlList.current();
+  }
 
-  public String getFirstUrl() {
-    return url != null  && !url.isEmpty() ? url.iterator().next() : null;
+  public void markAsUnavailable() {
+    urlList.fail(urlList.current());
```

Comment:
This is incorrect the url should be extracted from the exception and be passed through to this method.